app/controllers/users_controller.rb

```diff
@@ -1,7 +1,66 @@
 class UsersController < ApplicationController
 
-  def list
+  def show
     @users = User.all
   end
-  alias :index :list
+  alias :index :show
+
+  def new
+  end
+
+  def create
+    @user = User.create(user_params)
+    redirect_to '/'
+  end
+
+  def trim
+    user = User.find(params[:id])
+    user.update_attribute(:trimmed, true)
+    user.save
+    render :json => user
+  end
+
+  def massage_and_trim
+    user = User.find(params[:id])
+    user.update_attributes(:trimmed => true, :massaged => true)
+    user.save
+    render :json => user
+  end
+
+  def shave
+    user = User.find(params[:id])
+    user.update_attribute(:shaved, true)
+    user.save
```

Comment:
Be careful of usage of the non-bang version of .save unless you are going to operate on the return value.  Here you must use .save! in order to provide the correct HTTP response if something fails.

https://priyankapathak.wordpress.com/2010/08/20/usage-of-bang-in-ruby-on-rails/